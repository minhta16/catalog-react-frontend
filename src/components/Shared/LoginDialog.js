import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  DialogTitle,
  Dialog,
  DialogContent,
  TextField,
  DialogActions,
  Button,
  Link,
} from '@material-ui/core';
import { Link as RouterLink } from 'react-router-dom';

export class LoginDialog extends Component {
  state = {
    username: '',
    password: '',
  };

  onClickSubmit = () => {
    const { onClick, onClose } = this.props;
    const { username, password } = this.state;
    onClick(username, password);
    onClose();
  };

  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value,
    });
  };

  render() {
    const { open, onClose } = this.props;
    const { username, password } = this.state;
    return (
      <Dialog open={open} onClose={onClose}>
        <DialogTitle>Login</DialogTitle>
        <DialogContent>
          <form autoComplete="off">
            <TextField
              required
              id="username"
              label="Username"
              value={username}
              onChange={this.handleChange}
              margin="normal"
              fullWidth
              variant="outlined"
            />
            <TextField
              required
              id="password"
              label="Password"
              value={password}
              onChange={this.handleChange}
              type="password"
              margin="normal"
              fullWidth
              variant="outlined"
            />
          </form>
          <Link onClick={onClose} component={RouterLink} exact="true" to="/register">
            Havenâ€™t got an account yet? Register here!
          </Link>
        </DialogContent>
        <DialogActions>
          <Button id="login-dialog-login" onClick={this.onClickSubmit} color="primary">
            Login
          </Button>
          <Button id="login-dialog-close" onClick={onClose}>
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

LoginDialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  onClick: PropTypes.func,
};

LoginDialog.defaultProps = {
  onClose: () => {},
  onClick: async (username, password) => username + password,
  open: true,
};
export default LoginDialog;
