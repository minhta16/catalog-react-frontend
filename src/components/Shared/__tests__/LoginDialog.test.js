/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import LoginDialog from 'components/Shared/LoginDialog';
import { TextField } from '@material-ui/core';

describe('components/Shared/LoginDialog', () => {
  let wrapper;
  let props;
  // let warning;

  const setup = () => {
    wrapper = shallow(<LoginDialog {...props} />);
  };

  beforeEach(() => {
    props = {
      open: true,
      onClose: jest.fn(),
      onClick: jest.fn(),
      error: false,
      errorMessage: '',
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain 2 TextFields', () => {
    setup();
    expect(wrapper.find(TextField).length).toBe(2);
  });

  it('should call onClickSubmit -> onClose when submitted', () => {
    setup();
    wrapper.instance().onClickSubmit();
    expect(props.onClick).toHaveBeenCalled();
  });

  it('should call onClose when cancel is clicked', () => {
    setup();
    const { onClose } = props;
    const cancelButton = wrapper.find('#login-dialog-close');
    cancelButton.simulate('click');
    expect(onClose).toHaveBeenCalled();
  });

  it('should setState when a textfield is modified', () => {
    setup();
    const usernameText = wrapper.find('#username');
    usernameText.simulate('change', { target: { id: 'username', value: 'yolo' } });
    expect(wrapper.state().username).toBe('yolo');
  });

  it('should return username_password when default onClick is called ', () => {
    setup();
    LoginDialog.defaultProps.onClick('meomeo', 'mimi').then((retVal) => {
      expect(retVal).toBe('meomeo_mimi');
    });
  });

  it('should render error Typography when errorMessage is true', () => {
    props.errorMessage = 'this is an error!';
    setup();
    expect(wrapper.find('[color="error"]').length).toBe(1);
  });

  it('should not render error Typography when error is false', () => {
    props.error = false;
    setup();
    expect(wrapper.find('[color="error"]').length).toBe(0);
  });
});
