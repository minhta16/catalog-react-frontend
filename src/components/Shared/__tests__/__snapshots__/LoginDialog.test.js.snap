// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoginDialog should render correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Login
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <form
      autoComplete="off"
    >
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="username"
        label="Username"
        margin="normal"
        onChange={[Function]}
        required={true}
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
        variant="outlined"
      />
    </form>
    <WithStyles(ForwardRef(Link))
      component={[Function]}
      exact="true"
      onClick={[MockFunction]}
      to="/register"
    >
      Havenâ€™t got an account yet? Register here!
    </WithStyles(ForwardRef(Link))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      id="login-dialog-login"
      onClick={[Function]}
    >
      Login
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      id="login-dialog-close"
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
